src/private_cuda.h

```diff
@@ -128,4 +140,14 @@ struct _gpukernel {
 #endif
 };
 
+struct _gpucomm {
+  cuda_context* ctx;  // Start after the context
+// #ifdef WITH_CUDA_NCCL
+  ncclComm_t c;
+// #endif  // WITH_CUDA_NCCL
```

Comment:
For now this doesn't compile without nccl.  It's ok for development, but think about some way to not have variable-witdh structures that will also work without nccl.